app/controllers/CyaMaintainProfileController.scala

```diff
@@ -141,24 +141,17 @@ class CyaMaintainProfileController @Inject() (
 
   def onPageLoadNirmsNumber(): Action[AnyContent] =
     (identify andThen getData andThen requireData).async { implicit request =>
-      (for {
-        nirmsNumberOpt <- traderProfileConnector.getTraderProfile(request.eori).map(_.nirmsNumber)
-        userAnswers    <- nirmsNumberOpt.fold(Future.successful(request.userAnswers)) { nirmsNumber =>
-                            Future.fromTry(request.userAnswers.set(NirmsNumberUpdatePage, nirmsNumber))
-                          }
-      } yield userAnswers).flatMap { userAnswers =>
-        TraderProfile.getOptionallyRemovedPage(userAnswers) match {
-          case Right(_)     =>
-            val list = SummaryListViewModel(
-              rows = Seq(
-                HasNirmsSummary.rowUpdate(userAnswers),
-                NirmsNumberSummary.rowUpdate(userAnswers)
-              ).flatten
-            )
-            Future.successful(Ok(view(list, routes.CyaMaintainProfileController.onSubmitNirmsNumber)))
-          case Left(errors) =>
-            Future.successful(logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors))
-        }
+      TraderProfile.getOptionallyRemovedPage(request.userAnswers) match {
```

Comment:
This is an undoing of Ewan's last change, moved the idea to the RemoveNirmsController so everywhere can access the answer - that is the point where the answer becomes important